app/game/threat.js

```diff
@@ -2,23 +2,46 @@ const THREAT_TTL = 30;
 const THREAT_TICK_SPEED = 1000;
 const THREAT_COOLDOWN_SECONDS = 5;
 
-const Threat = (threatType, onThreatUnresolved) => {
+const THREAT_RESOLVE_TIME_MS = 3*1000; // How long it takes when a player enters a room w/ the right tool for the threat to disappear
+
+const Threat = (threatType, onThreatUnresolved, onThreatResolved) => {
     let currentAge = THREAT_TTL;
     const THREAT_TYPE = threatType;
+    let resolved = false;
+    let isResolving = false;
+    const MAP_THREAT_TO_TOOL = {
+        "fire" : "fire-extinguisher",
+        "breach" : "wrench",
+        "invader" : "gun"
+    }
```

Comment:
Let's move this constant map outside the Threat object, since they all share this